src/isar/services/readers/base_reader.py

```
@@ -47,13 +47,17 @@ def dict_to_dataclass(
             logger.error(
                 f"A type of a input value does not match with a type of a data class field {e}"
             )
-            return None
+            raise BaseReaderError
         except MissingValueError as e:
             logger.error(f"A value for a required field is not provided {e}")
-            return None
+            raise BaseReaderError
         except ValueError as e:
             logger.error({e})
-            return None
+            raise BaseReaderError
```

Why not propagate the same error farther?